packages/coinbase/src/index.ts

```diff
@@ -31,29 +45,42 @@ function coinbaseWallet({
             ? (CoinbaseWalletSDK as any).default
             : CoinbaseWalletSDK
         ) as typeof CoinbaseWalletSDK
+        const { isHex, toHex } = await import('@web3-onboard/common')
 
         const base64 = window.btoa(icon || '')
         const appLogoUrl = `data:image/svg+xml;base64,${base64}`
 
+        const appChainIds = chains.map(
+          // @ts-ignore - treating hex strings as numbers as they are expected to be hex numbers
+          ({ id }) => id as number
+        )
+
         const instance = new CoinbaseWalletSDKConstructor({
           appName: name || '',
           appLogoUrl,
-          darkMode,
-          enableMobileWalletLink,
-          reloadOnDisconnect
+          appChainIds
         })
 
-        const coinbaseWalletProvider = instance.makeWeb3Provider(
-          chain.rpcUrl,
-          parseInt(chain.id)
-        )
+        const coinbaseWalletProvider = instance.makeWeb3Provider({
+          options: supportedWalletType
+        })
 
         // patch the chainChanged event
         const on = coinbaseWalletProvider.on.bind(coinbaseWalletProvider)
+
         coinbaseWalletProvider.on = (event, listener) => {
+          // @ts-ignore
           on(event, val => {
             if (event === 'chainChanged') {
-              listener(`0x${(val as number).toString(16)}`)
+              let hexVal: string
+              if (isHex(val)) {
+                hexVal = val
+              } else {
+                hexVal = toHex(val)
+              }
+
+              // @ts-ignore
+              listener(hexVal)
```

Comment:
Thank you for handling this. We will update the `chainChanged` event in the ext and mobile providers to make them EIP-1193 compliant. Appreciate the workaround in the meantime 🙏